src/OpenConext/ProfileBundle/Service/GlobalViewParameters.php

```diff
@@ -0,0 +1,159 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\ProfileBundle\Service;
+
+use InvalidArgumentException;
+use OpenConext\Profile\Assert;
+use Symfony\Component\Translation\DataCollectorTranslator;
+
+class GlobalViewParameters
+{
+    /**
+     * @var DataCollectorTranslator
+     */
+    private $translator;
+
+    /**
+     * @var array
+     */
+    private $helpUrls;
+
+    /**
+     * @var array
+     */
+    private $platformUrls;
+    /**
+     * @var array
+     */
+    private $termsOfServiceUrls;
+
+    /**
+     * @param DataCollectorTranslator $translator
+     * @param array $helpUrls
+     * @param array $platformUrls
+     * @param array $termsOfServiceUrls
+     */
+    private function __construct(
+        DataCollectorTranslator $translator,
+        array $helpUrls,
+        array $platformUrls,
+        array $termsOfServiceUrls
+    ) {
+
+        $this->translator = $translator;
+        $this->helpUrls = $helpUrls;
+        $this->platformUrls = $platformUrls;
+        $this->termsOfServiceUrls = $termsOfServiceUrls;
+    }
+
+    /**
+     * @param DataCollectorTranslator $translator
+     * @param array $locales
+     * @param array $helpUrls
+     * @param array $platformUrls
+     * @param array $termsOfServiceUrls
+     * @return GlobalViewParameters
+     */
+    public static function createForLocales(
+        DataCollectorTranslator $translator,
+        array $locales,
+        array $helpUrls,
+        array $platformUrls,
+        array $termsOfServiceUrls
+    ) {
+        if (!in_array($translator->getLocale(), $locales)) {
+            throw new InvalidArgumentException(sprintf(
+                'Locale "%s" is not configured as a valid locale. Currently configured locales: %s',
+                $translator->getLocale(),
+                implode(', ', $locales)
+            ));
+        }
+
+        foreach ($locales as $locale) {
+            Assert::keyExists($helpUrls, $locale, sprintf('Locale "%s" not configured for helpUrls', $locale));
+            Assert::keyExists($platformUrls, $locale, sprintf('Locale "%s" not configured for platformUrls', $locale));
+            Assert::keyExists($termsOfServiceUrls, $locale, sprintf(
+                'Locale "%s" not configured for termsOfServiceUrls',
+                $locale
+            ));
+        }
+
+        self::checkIfNoExtraLocales($helpUrls, $locales, 'helpUrls');
+        self::checkIfNoExtraLocales($platformUrls, $locales, 'platformUrls');
+        self::checkIfNoExtraLocales($termsOfServiceUrls, $locales, 'termsOfServiceUrls');
```

Comment:
Checking their count would suffice. But that would be less explicit as it would depend on said additional test. In this case, that probably would not be a problem.